### PR TITLE
Set sysctl/Namespace kernel parameters within the container

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

**Comment by rhatdan** [comment]
This is needed for https://github.com/docker/docker/pull/19265

**Comment by MHBauer** [comment]
I understand what the `omitempty` annotation does. What does it mean in this specific case of an empty map vs it not being there? Does make the code easier on one side or the other? I'm still learning how json is used in the golang context.

I don't have any problems adding a field if it's going to be used.

**Comment by thaJeztah** [comment]
ping @rhatdan can you rebase?

@calavera the associated PR in docker/docker is now in code-review

**Comment by rhatdan** [comment]
Done

**Comment by calavera** [comment]
LGTM

File: types/container/host_config.go [hunk 1]
```diff
@@ -231,6 +231,7 @@ type HostConfig struct {
 	Tmpfs           map[string]string `json:",omitempty"` // List of tmpfs (mounts) used for the container
 	UTSMode         UTSMode           // UTS namespace to use for the container
 	ShmSize         int64             // Total shm memory usage
+	Sysctls         map[string]string `json:",omitempty"` // List of Namesp
```
Comment (thaJeztah) [inline types/container/host_config.go]: oh, nitpick but does the comment match this? "List of Namesp"

**Comment by thaJeztah** [comment]
@rhatdan looks like you did a merge, not a rebase

**Comment by rhatdan** [comment]
@thaJeztah I think it is fixed now

**Comment by thaJeztah** [comment]
Thanks @rhatdan LGTM

ping @calavera for the merge :smile:

**Comment by runcom** [comment]
LGTM

**Pull request merged**